quiche/src/recovery/bbr/mod.rs

```diff
@@ -255,13 +255,14 @@ impl State {
 fn bbr_enter_recovery(r: &mut Recovery, now: Instant) {
     r.bbr_state.prior_cwnd = per_ack::bbr_save_cwnd(r);
 
-    r.congestion_window = r.bytes_in_flight +
-        r.bbr_state.newly_acked_bytes.max(r.max_datagram_size);
+    r.congestion_window = r.bytes_in_flight.max(r.max_datagram_size);
     r.congestion_recovery_start_time = Some(now);
 
     r.bbr_state.packet_conservation = true;
     r.bbr_state.in_recovery = true;
 
+    r.bbr_state.newly_lost_bytes = 0;
```

Comment:
`newly_lost_bytes` is used in [ack processing](https://github.com/cloudflare/quiche/blob/master/quiche/src/recovery/bbr/per_ack.rs#L164), which is called in [`on_packets_acked()` callback and cleared right after.] (https://github.com/cloudflare/quiche/blob/master/quiche/src/recovery/bbr/mod.rs#L320-L322)
If we clear it here, it won't be used I think

Reply:
Probably we need to understand what tcp (or other) does. their in_flight is including lost packets already? (if not, the algorithm is correct)

I understand lost_bytes can be used twice. If so, instead of doing L264, we need to remove https://github.com/cloudflare/quiche/blob/master/quiche/src/recovery/bbr/per_ack.rs#L166-L171 . lost_bytes is used in other places, so I don't think just clearing up here is not a good idea. I'll update my PR (#1478) to reflect that.

Reply:
I meant to say, lost_bytes are counted twice only when we enter the recovery episode, not when we are in recovery. 
https://github.com/cloudflare/quiche/blob/master/quiche/src/recovery/bbr/per_ack.rs#L166-L171 is needed because we may see losses while we are in recovery.

Reply:
Ahhh right... Then I think we need to add lost_bytes back in bbr_enter_recovery().
See if it makes sense:
https://github.com/cloudflare/quiche/pull/1478/files#diff-89fb0e46580c044508e7c118b4124b0f151e2a4437e66029fb897cdb18c50a12R263

Reply:
The current is doing the same thing right ? Why do you want to add and then subtract ?

Reply:
When we are entering recovery lost_bytes is made zero since itss already calculated in in_flight. In recovery phase we are not making lost_bytes zero so it is counted.